src/kofta_metadata.erl

```diff
@@ -254,16 +276,36 @@ make_request(State) ->
 
     TopicNames = dict:fetch_keys(ClientDict),
 
-    NewState = State#st{clients=dict:new(), last_batch=now()},
+    NewState = State#st{
+        clients=dict:new(),
+        broker_clients=dict:new(),
+        last_batch=now()
+    },
 
     case do_request(TopicNames, Host, Port) of
-        {ok, Topics} ->
-            lists:map(fun(Topic) ->
-                Clients = dict:fetch(Topic#topic.name, ClientDict),
-                lists:map(fun(Client) ->
-                    gen_fsm:reply(Client, {ok, Topic})
-                end, Clients)
-            end, Topics),
+        {ok, Brokers, Topics} ->
+            case dict:size(ClientDict) of
+                0 ->
```

Comment:
If ClientDict is empty, it means that the request was executed with no topics supplied (ie. you just asked for the broker info without any topics). And kafka will reply with the metadata for all topics if you don't supply any topics in the request. Otherwise kafka will reply with topic info for each topic in your request.